pkg/endpoint/policy.go

```diff
@@ -529,7 +529,18 @@ func (e *Endpoint) regenerate(owner Owner) error {
 		return fmt.Errorf("Failed to create endpoint directory: %s", err)
 	}
 
-	if err := e.regenerateBPF(owner, tmpDir); err != nil {
+	defer func() {
+		// Set to Ready, but only if no other changes are pending.
```

Comment:
I thought I understood the reason when I approved this but when Andre asked me today I could not recall. can you clarify this comment to explain why we need to go into ready state even on failure?

Reply:
"ready" state is reported by the CLI as "not-ready" of the Status is not OK. Note that this PR also removes the wrong setting of Status to OK in the failure case.

Also, we do try to roll back to the previous policy, even though the bpf map changes are not currently rolled back.

current behavior is to remain in the regenerating state until a future change leads to successful regeneration. This PR does not change the end result in that case, as successful regeneration will also set the status to OK, allowing CLI to report the state as "ready".

Reply:
This was accompanied by a CLI feature that policy wait does wait forever. This I addressed in a separate PR.